orders/views.py

```diff
@@ -11,15 +11,72 @@
     Order,
     OrderStatus,
     OrderItem,
-    ShippingStatus,
     OrderStatusEnum,
+    ShippingStatus,
     ShipptingStatusEnum
     )
 from users.utils     import login_decorator
 from products.models import ProductOption
 
 class OrderView(View):
     @login_decorator
+    @transaction.atomic
+    def post(self, request):
+        user = request.user
+
+        product_options = ProductOption.objects.filter(cart__user=user)\
+                                               .prefetch_related('cart_set')\
+                                               .annotate(total_quantity=Sum('cart__quantity'))
+
+        sold_out = [
+            {
+                'product_name': product_option.product.name
+            }
+            for product_option in product_options 
+            if product_option.stock - product_option.cart_set.get(user=user).quantity < 0
+        ]
+        
+        if sold_out: 
+            return JsonResponse(
+                    {
+                        'message' : 'OUT_OF_STOCK',
+                        'sold_out': sold_out
+                    }, status=202)
```

Comment:
suggestion 처럼 좀 더 깔끔하게 줄여서 작성해볼 수 있을 것 같네요.
그리고 위의 쿼리셋에서 만들어 둔 `total_quantity`를 활용하면 더 좋겠습니다.
```suggestion
        for product_option in product_options:
            if product_option.stock - product_option.total_quantity < 0:
                return JsonResponse({
                    'message' : 'OUT_OF_STOCK',
                    ...
                })
```